lib/crowbar/client/command/proposal/edit.rb

```diff
@@ -140,6 +150,29 @@ def payload_content
               from_editor
             end
           end
+
+          def check_nodes(configuration)
+            avaliable_nodes = JSON.parse(Request::Node::List.new({}).process)
+            if avaliable_nodes["nodes"].nil?
+              say "Internal error: Invalid node list provided"
+              raise
+            end
+
+            avaliable_nodes = avaliable_nodes["nodes"]
+
+            configuration["deployment"].each do |service_name, service|
+              next if service["elements"].nil?
+
+              service["elements"].each do |role, nodes|
+                nodes.each do |node|
+                  if avaliable_nodes.select { |item| item["name"] == node }.empty?
```

Comment:
... with the above change you can simplify this check: `unless available_nodes.include? node`